### PR TITLE
feat(plasma-hope,plasma-web,plama-b2c): Add `TextField` union

### PR DESCRIPTION
**Pull request by neretin-trike**
Необходимо было объединить компонент `TextField`. Теперь в библиотеках plasma-web и plasma-b2c реализация этого компонента и API (т.е. все его свойства) - одинаковая. 
В самом plasma-hope основная логика выделена в файл `TextField`, в котором в зависимости от выбранного свойства `design`, будет возвращаться вью компонента.

Итоговая структура компонента:
```
plasma-web // directory
|
└─ TextField // re-export with `design`
  |
  └─ plasma-hope //  directory
    |
    └─ TextField // business logic
       |
       └─ TextFieldB2c/Web // view
```

:baby_chick: Download canary assets:

[colors--canary.429.4508670921.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.429.4508670921.xml)  
[color_metro_ios-swift--canary.429.4508670921.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_ios-swift--canary.429.4508670921.swift)  
[color_metro_kotlin--canary.429.4508670921.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_kotlin--canary.429.4508670921.kt)  
[color_metro_react-native--canary.429.4508670921.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_react-native--canary.429.4508670921.ts)  
[color_sberHealth_ios-swift--canary.429.4508670921.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_ios-swift--canary.429.4508670921.swift)  
[color_sberHealth_kotlin--canary.429.4508670921.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_kotlin--canary.429.4508670921.kt)  
[color_sberHealth_react-native--canary.429.4508670921.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_react-native--canary.429.4508670921.ts)  
[color_sbermarket_ios-swift--canary.429.4508670921.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.429.4508670921.swift)  
[color_sbermarket_kotlin--canary.429.4508670921.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.429.4508670921.kt)  
[color_sbermarket_react-native--canary.429.4508670921.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.429.4508670921.ts)  
[color_sberprime_ios-swift--canary.429.4508670921.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.429.4508670921.swift)  
[color_sberprime_kotlin--canary.429.4508670921.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.429.4508670921.kt)  
[color_sberprime_react-native--canary.429.4508670921.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.429.4508670921.ts)  
[color_selgros_ios-swift--canary.429.4508670921.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_ios-swift--canary.429.4508670921.swift)  
[color_selgros_kotlin--canary.429.4508670921.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_kotlin--canary.429.4508670921.kt)  
[color_selgros_react-native--canary.429.4508670921.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_react-native--canary.429.4508670921.ts)  
[color_smbusiness_ios-swift--canary.429.4508670921.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_ios-swift--canary.429.4508670921.swift)  
[color_smbusiness_kotlin--canary.429.4508670921.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_kotlin--canary.429.4508670921.kt)  
[color_smbusiness_react-native--canary.429.4508670921.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_react-native--canary.429.4508670921.ts)  
[PlasmaTokensColor--canary.429.4508670921.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.429.4508670921.swift)  
[shadow_sbermarket_react-native--canary.429.4508670921.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.429.4508670921.ts)  
[typo_mage_ios-swift--canary.429.4508670921.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.429.4508670921.swift)  
[typo_mage_kotlin--canary.429.4508670921.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.429.4508670921.kt)  
[typo_mage_react-native--canary.429.4508670921.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.429.4508670921.ts)  
[typo_plasma_ios-swift--canary.429.4508670921.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.429.4508670921.swift)  
[typo_plasma_kotlin--canary.429.4508670921.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.429.4508670921.kt)  
[typo_plasma_react-native--canary.429.4508670921.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.429.4508670921.ts)  
[typo_ruler_ios-swift--canary.429.4508670921.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.429.4508670921.swift)  
[typo_ruler_kotlin--canary.429.4508670921.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.429.4508670921.kt)  
[typo_ruler_react-native--canary.429.4508670921.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.429.4508670921.ts)  
[typo_sage_ios-swift--canary.429.4508670921.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.429.4508670921.swift)  
[typo_sage_kotlin--canary.429.4508670921.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.429.4508670921.kt)  
[typo_sage_react-native--canary.429.4508670921.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.429.4508670921.ts)  
[typo_sbermarket_ios-swift--canary.429.4508670921.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.429.4508670921.swift)  
[typo_sbermarket_kotlin--canary.429.4508670921.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.429.4508670921.kt)  
[typo_sbermarket_react-native--canary.429.4508670921.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.429.4508670921.ts)  
[typo_soulmate_ios-swift--canary.429.4508670921.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.429.4508670921.swift)  
[typo_soulmate_kotlin--canary.429.4508670921.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.429.4508670921.kt)  
[typo_soulmate_react-native--canary.429.4508670921.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.429.4508670921.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.165.0-canary.429.4508670921.0
  npm install @salutejs/plasma-hope@0.23.0-canary.429.4508670921.0
  npm install @salutejs/plasma-web@1.190.0-canary.429.4508670921.0
  # or 
  yarn add @salutejs/plasma-b2c@1.165.0-canary.429.4508670921.0
  yarn add @salutejs/plasma-hope@0.23.0-canary.429.4508670921.0
  yarn add @salutejs/plasma-web@1.190.0-canary.429.4508670921.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-429/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-429/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-429/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-429/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-429/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-429/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

File: packages/plasma-hope/src/components/TextField/types.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    /**
+     * @deprecated внутреннее свойство.
+     */
+    $isFocused?: boolean;
+    /**
+     * @deprecated устаревшее свойство.
+     */
+    htmlSize?: InputHTMLAttributes<HTMLInputElement>['size'];
+    /**
+     * Доступные размеры компонента. (для plasma-web: l, m, s).
```
Comment (Yakutoc) [inline packages/plasma-hope/src/components/TextField/types.ts:18]: [Уточнение]:

Мы хотим что пользователи видели эти `(для plasma-web: l, m, s).` уточнения?

Снимок экрана 2023-03-20 в 16 51 58

**Review by Yakutoc — APPROVED** [review]

File: packages/plasma-hope/src/components/TextField/types.ts (hunk 1 above)
Comment (neretin-trike) [reply packages/plasma-hope/src/components/TextField/types.ts:18]: ну скорее всего да, т.к. хотим, чтобы интерфейс у обоих библиотек был полностью идентичен

File: packages/plasma-hope/src/components/TextField/views/web/Input.props.tsx [hunk 2]
```diff
@@ -44,6 +44,16 @@ export const sizes = {
         hasContentPaddingLeft: 'var(--plasma-input-s-has-content-padding-left)',
         hasContentPaddingRight: 'var(--plasma-input-s-has-content-padding-right)',
     },
+    // INFO: Временно s = xs, пока не произведен редизайн
```
[the remark this answers is not in the document]
Comment (neretin-trike) [reply packages/plasma-hope/src/components/TextField/views/web/Input.props.tsx:47]: ишью отсутствует, но это скорее именно информация. Возможно при редизайне этот файл вообще уйдёт

**Review by kayman233 — APPROVED** [review]

File: packages/plasma-b2c/src/components/TextField/TextField.tsx [hunk 3]
```diff
@@ -1,137 +1,7 @@
...
-    view?: TextFieldView;
-
-    /**
-     * Callback по нажатию Enter
-     */
-    onSearch?: (value: string, event?: KeyboardEvent<HTMLInputElement>) => void;
-}
-
-/**
- * Поле ввода текста.
```
Comment (Yeti-or) [inline packages/plasma-b2c/src/components/TextField/TextField.tsx:42]: вот это потеряли, не нужно ?

**Review by Yeti-or — APPROVED** [review]

File: packages/plasma-b2c/src/components/TextField/TextField.tsx (hunk 3 above)
Comment (neretin-trike) [reply packages/plasma-b2c/src/components/TextField/TextField.tsx:42]: эх, хотелось, чтобы он это описание подхватил у вьюх, но похоже нет, поправлю, спасиб 👌

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-429/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-429/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-429/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-429/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-429/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-429/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-429/temple-storybook/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-429/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-429/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-429/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-429/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-429/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-429/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-429/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Pull request merged**